packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -73,3 +89,65 @@ Learn more about compiler configuration at https://usehardhat.com/configuration"
 
   return resolved;
 }
+
+/**
+ * Receives a parsed stack trace of a MODULE_NOT_FOUND error and checks
+ * if the error comes from a missing peer dependency from a hardhat plugin.
+ *
+ * If that's the case, it throws an error. Otherwise it does nothing.
+ */
+export function analyzeModuleNotFoundStackTrace(
+  stackTrace: stackTraceParser.StackFrame[],
+  configPath: string
+) {
+  const throwingFile = stackTrace
+    .filter((x) => x.file !== null)
+    .map((x) => x.file!)
+    .find((x) => path.isAbsolute(x));
+
+  if (throwingFile === null || throwingFile === undefined) {
+    return;
+  }
+
+  // if the error comes from the config file, we ignore it because we know it's
+  // a direct import that's missing
+  if (throwingFile === configPath) {
+    return;
+  }
+
+  const packageJsonPath = findUp.sync("package.json", {
```

Comment:
There's `findClosestPackageJson` in the `packageInfo` module.

Reply:
I had to export it to use it, but I think that's fine?